util/settings/settings.go

```diff
@@ -544,6 +548,14 @@ func (mgr *SettingsManager) GetTrackingMethod() (string, error) {
 	return argoCDCM.Data[settingsResourceTrackingMethodKey], nil
 }
 
+func (mgr *SettingsManager) GetInClusterServerAddressAllowed() (bool, error) {
+	argoCDCM, err := mgr.getConfigMap()
+	if err != nil {
+		return true, err
+	}
+	return argoCDCM.Data[inClusterServerAddressAllowedKey] == "true", nil
+}
+
```

Comment:
For simple settings fields that are just strings and don't need yaml marshalling, we don't need to use accessor methods. So it is okay to delete this method and the code will rely on `settings.clusterInClusterEnabled`.